pkg/deployer/deployer.go

```diff
@@ -257,22 +258,39 @@ func (d *DeployerImpl) deploy(
 }
 
 // GetDeployments returns deployments from a map of nodes IDs and deployments IDs
-func (d *DeployerImpl) GetDeployments(ctx context.Context, sub subi.SubstrateExt, dls map[uint32]uint64) (map[uint32]gridtypes.Deployment, error) {
-	res := make(map[uint32]gridtypes.Deployment)
+func (d *DeployerImpl) GetDeployments(ctx context.Context, sub subi.SubstrateExt, dls map[uint32]uint64) (res map[uint32]gridtypes.Deployment, err error) {
+	res = make(map[uint32]gridtypes.Deployment)
+
+	var wg sync.WaitGroup
+	var mux *sync.RWMutex
+
 	for nodeID, dlID := range dls {
-		nc, err := d.ncPool.GetNodeClient(sub, nodeID)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to get a client for node %d", nodeID)
-		}
-		sub, cancel := context.WithTimeout(ctx, 10*time.Second)
-		defer cancel()
-		dl, err := nc.DeploymentGet(sub, dlID)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to get deployment %d of node %d", dlID, nodeID)
-		}
-		res[nodeID] = dl
+
+		wg.Add(1)
+		go func(nodeID uint32, dlID uint64) {
+
+			nc, clientErr := d.ncPool.GetNodeClient(sub, nodeID)
+			if clientErr != nil {
+				err = errors.Wrapf(clientErr, "failed to get a client for node %d", nodeID)
+			}
+
+			sub, cancel := context.WithTimeout(ctx, 10*time.Second)
+			defer cancel()
+
+			dl, clientErr := nc.DeploymentGet(sub, dlID)
+			if clientErr != nil {
+				err = errors.Wrapf(clientErr, "failed to get deployment %d of node %d", dlID, nodeID)
+			}
+
+			mux.Lock()
+			res[nodeID] = dl
+			mux.Unlock()
+
+		}(nodeID, dlID)
 	}
-	return res, nil
+
+	wg.Wait()
+	return
```

Comment:
maybe remove the named return parameters and not use the naked return, that probably was the reason we switched from err name to `clientErr`? we can also collect the errors using multierror if needed, also the function should return an empty map in case of an error in the final result